controller/src/main/java/io/pravega/controller/store/stream/InMemoryBucketStore.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import io.pravega.controller.server.bucket.BucketChangeListener;
+import io.pravega.controller.server.bucket.BucketOwnershipListener;
+import lombok.Getter;
+import lombok.Synchronized;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executor;
+
+public class InMemoryBucketStore implements BucketStore {
+    @Getter
+    private final int bucketCount;
+    
+    private final ConcurrentMap<String, Set<String>> bucketedStreams;
+    
+    private final ConcurrentMap<String, BucketChangeListener> listeners;
+
+    InMemoryBucketStore(int bucketCount) {
+        this.bucketCount = bucketCount;
+        listeners = new ConcurrentHashMap<>();
+        bucketedStreams = new ConcurrentHashMap<>();
+    }
+
+
+    @Override
+    public void registerBucketChangeListener(ServiceType serviceType, int bucket, BucketChangeListener listener) {
+        listeners.put(getBucketName(serviceType, bucket), listener);
+    }
+
+    private String getBucketName(ServiceType serviceType, int bucket) {
+        return serviceType.getName() + "/" + bucket;
+    }
+
+    @Override
+    public void unregisterBucketChangeListener(ServiceType serviceType, int bucket) {
+        listeners.remove(getBucketName(serviceType, bucket));
+    }
+
+    @Override
+    public void registerBucketOwnershipListener(ServiceType serviceType, BucketOwnershipListener ownershipListener) {
+    }
```

Comment:
Are these unimplemented on purpose? Then should they throw `UnsupportedOperationException`?

Reply:
These are implemented with a default No-Op implementation